spring-integration-core/src/main/java/org/springframework/integration/config/annotation/InboundChannelAdapterAnnotationPostProcessor.java

```diff
@@ -72,7 +74,16 @@ public Object postProcess(Object bean, String beanName, Method method, List<Anno
 			return null;
 		}
 
-		MessageChannel channel = this.channelResolver.resolveDestination(channelName);
+		MessageChannel channel;
+		try {
+			channel = this.channelResolver.resolveDestination(channelName);
+		}
+		catch (DestinationResolutionException e) {
+			channel = new DirectChannel();
+			this.beanFactory.registerSingleton(channelName, channel);
+			channel = (MessageChannel) this.beanFactory.initializeBean(channel, channelName);
+		}
+		Assert.notNull(channel, "failed to resolve channel '" + channelName + "'");
```

Comment:
I wonder if it would be better to inject the channel name into the SPCA and resolve it there at runtime.

Perhaps the "yet-to-be-declared" channel might not be a `DirectChannel`.

Thoughts?